src/plugins/blocksmith/components/imgur-post-block-modal/index.jsx

```diff
@@ -0,0 +1,207 @@
+  }
+
+  handleChangeDate(date) {
+    this.setState(changedDate(date));
+  }
+
+  handleChangeTime({ target: { value: time } }) {
+    this.setState(changedTime(time));
+  }
+
+  handleChangeCheckbox({ target: { id, checked } }) {
+    this.setState({ [id]: checked });
+  }
+
+  handleChangeTextArea(event) {
+    const input = event.target.value;
+    const id = getImgurPostMediaId(input);
+    let { errorMsg, imgurId, isValid } = this.state;
+
+    if (!id) {
+      errorMsg = 'url or embed code is invalid';
+      imgurId = input;
+    } else {
+      errorMsg = '';
+      imgurId = id;
+      isValid = true;
+    }
+    this.setState({
+      errorMsg,
+      imgurId,
+      isValid,
+      touched: true,
+    });
+  }
+
+  render() {
+    const {
+      aside,
+      errorMsg,
+      hasUpdatedDate,
+      showContext,
+      isValid,
+      touched,
+      imgurId,
+      updatedDate,
+    } = this.state;
+    const { isFreshBlock, isOpen, toggle } = this.props;
+
+    return (
+      <Modal isOpen={isOpen} toggle={toggle}>
+        <ModalHeader toggle={toggle}>{`${isFreshBlock ? 'Add' : 'Update'} Imgur Post Block`}</ModalHeader>
+        <ModalBody>
+          <FormGroup>
+            <Label>{isValid ? 'Post id' : 'Embed code, Link, URI, or Post id'}</Label>
+            <Input
+              innerRef={(el) => { this.inputElement = el; }}
+              onChange={this.handleChangeTextArea}
+              placeholder="enter embed code"
+              type="textarea"
+              value={imgurId || ''}
+            />
+          </FormGroup>
+          {
+            !isValid && touched && imgurId !== ''
+            && <p className="text-danger">{errorMsg}</p>
+          }
+
+          <FormGroup>
+            <Checkbox size="sd" id="hasUpdatedDate" checked={hasUpdatedDate} onChange={this.handleChangeCheckbox}>
+              Is update
+            </Checkbox>
+          </FormGroup>
+
+          <FormGroup>
+            <Checkbox size="sd" id="aside" checked={aside} onChange={this.handleChangeCheckbox}>
+              Aside
+            </Checkbox>
+            <Icon imgSrc="info-outline" id="aside-tooltip" size="xs" className="ml-1" />
+            <UncontrolledTooltip target="aside-tooltip">Is only indirectly related to the main content.</UncontrolledTooltip>
+          </FormGroup>
+
+          <FormGroup>
+            <Checkbox size="sd" id="showContext" checked={showContext} onChange={this.handleChangeCheckbox}>
+            Show title
+            </Checkbox>
+          </FormGroup>
+          {hasUpdatedDate
+            && (
+              <div className="modal-body-blocksmith">
+                <DateTimePicker
+                  onChangeDate={this.handleChangeDate}
+                  onChangeTime={this.handleChangeTime}
+                  updatedDate={updatedDate}
+                />
+              </div>
+            )}
+          {
+            imgurId && <ImgurPostBlockPreview key={Math.random()} block={this.setBlock()} />
```

Comment:
@jalopez13 do you still need this key?

Reply:
@efjacobson 

We need the key so when the show title toggle is checked it re-renders the preview. Don't work without.

Reply:
@jalopez13 aha - the key makes it work because we are not doing a bunch of dom manipulation like i originally had thought we would need to. its fine, but please leave a comment:

`the key makes it so we can avoid doing a ton of dom manipulation to reconcile what the embed script does to the dom`

Reply:
@efjacobson 

I'll add a comment.